src/h5web/vis-packs/core/heatmap/MappedHeatmapVis.tsx

```diff
@@ -40,24 +38,16 @@ function MappedHeatmapVis(props: Props): ReactElement {
   const baseArray = useBaseArray(value, dims);
   const dataArray = useMappedArray(baseArray, dimensionMapping);
 
-  const dataDomain = useDomain(
-    (autoScale ? dataArray.data : baseArray.data) as number[],
-    scaleType
-  );
-  const prevDataDomain = usePrevious(dataDomain);
-
-  // Disable `autoScale` for 2D datasets (baseArray and dataArray span the same values)
-  useEffect(() => {
-    disableAutoScale(baseArray.shape.length <= 2);
-  }, [baseArray.shape, disableAutoScale]);
-
-  // Use `customDomain` if any, unless `dataDomain` just changed (in which case it is stale and needs to be reset - cf. `useEffect` below)
-  const domain =
-    customDomain && dataDomain === prevDataDomain ? customDomain : dataDomain;
+  const domain = useMemo(() => {
+    return customDomain || getDomain(dataArray.data as number[], scaleType);
+  }, [customDomain, dataArray, scaleType]);
 
   useEffect(() => {
-    resetDomains(dataDomain); // in config, update `dataDomain` and reset `customDomain`
-  }, [dataDomain, resetDomains]);
+    if (!customDomain) {
+      // If auto-scale is on, update `dataDomain`
+      setDataDomain(domain);
+    }
+  }, [customDomain, domain, setDataDomain]);
```

Comment:
A LOT simpler logic overall 😂

The `useMemo` allows not computing the `dataDomain` if we don't need to.

Reply:
Nice !